sale_blanket_order/models/planned_blanket_orders.py

```diff
@@ -0,0 +1,73 @@
+# coding: utf-8
+# Copyright 2018 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import fields, models, api, _
+from odoo.exceptions import UserError
+
+
+class BlanketPlannedOrder(models.Model):
+    _name = 'sale.blanket.planned.order'
+    _description = 'Blanket Planned Order'
+    blanket_order_id = fields.Many2one(
+        'sale.blanket.order', required=True, ondelete='cascade')
+    next_order_date = fields.Date(string='next order date', required=True)
+    line_ids = fields.One2many(
+        'sale.blanket.planned.order.line', 'planned_order_id', string='Order lines',
+        copy=True)
+
+    @api.multi
+    def create_order(self):
+        for rec in self:
+            if fields.Date.today() == rec.next_order_date:
+                line_obj = self.env['sale.order.line']
+                order_lines = []
+                for line in rec.line_ids:
+
+                    vals = {
+                        'product_id': line.product_id.id,
+                        'product_uom': line.blanket_order_line_id.product_uom.id,
+                        'sequence': line.blanket_order_line_id.sequence,
+                        'price_unit': line.blanket_order_line_id.price_unit,
+                        'blanket_line_id': line.blanket_order_line_id.id,
+                    }
+                    vals.update(line_obj.onchange(
+                        vals, 'product_id', {'product_id': 'true'})['value'])
+                    vals['product_uom_qty'] = line.next_order_qty
+                    order_lines.append((0, 0, vals))
+
+                if not order_lines:
+                    raise UserError(_('An order can\'t be empty'))
+
+                order_vals = {
+                    'partner_id': rec.blanket_order_id.partner_id.id,
+                }
+                order_vals.update(rec.env['sale.order'].onchange(
+                    order_vals, 'partner_id', {'partner_id': 'true'})['value'])
+                order_vals.update({
+                    'user_id': rec.blanket_order_id.user_id.id,
+                    'origin': rec.blanket_order_id.name,
+                    'currency_id': rec.blanket_order_id.currency_id.id,
+                    'order_line': order_lines,
+                    'pricelist_id': (rec.blanket_order_id.pricelist_id.id
+                                     if rec.blanket_order_id.pricelist_id
+                                     else False),
+                    'payment_term_id': (rec.blanket_order_id.payment_term_id.id
+                                        if rec.blanket_order_id.payment_term_id
+                                        else False),
+                })
+                self.env['sale.order'].create(order_vals)
```

Comment:
Define obj before loop and not inside the loop

Reply:
fixed!